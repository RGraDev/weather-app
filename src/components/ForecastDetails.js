import React from "react";
import iconData from "../data/iconData.json";

function ForecastDetails({ forecast }) {
  const { date, description, temperature, humidity, wind, icon } = forecast;

  const formattedDate = new Date(date).toDateString();
  const weatherCode = Math.floor(icon / 100) * 100;

  return (
    <div className="forecast=details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__description">{description}</div>
      <div className="forecast-details__icon">
        <img src={iconData[weatherCode]} alt="weather icon" />
      </div>
      <div className="forecast-details__min-temp">{temperature.min}&deg;C</div>
      <div className="forecast-details__max-temp">{temperature.max}&deg;C</div>
      <div className="forecast-details__humidity">{humidity}</div>
      <div className="forecast-details__wind-speed">{wind.speed}mph</div>
      <div className="forecast-details__wind-direction">{wind.direction.toUpperCase()}</div>
    </div>
  );
}
export default ForecastDetails;
