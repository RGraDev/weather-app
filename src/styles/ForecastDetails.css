.forecast-details {
    margin-top: 5vh;
    padding: 20px;
    margin-bottom: 40px;
    display: flex;
    flex-direction: column;
    align-items: center;
    border: 1px solid red;
}

.forecast-details__date {
    margin-bottom: 10px;
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;
    align-items: center;
    font-weight: bold;
}

.forecast-details__icon, .forecast-details__description {
    margin: 20px 0;
    width: 100%;
    display: flex;
    flex-flow: row nowrap;
    align-items: center;
    justify-content: center;
    font-weight: bold;
}
