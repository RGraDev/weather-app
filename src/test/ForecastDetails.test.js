import { render } from "@testing-library/react";
import ForecastDetails from "../components/ForecastDetails";

describe("ForecastDetails", function () {
  const validProps = {
    forecast: {
      date: 1525046400000,
      description: "stub description",
      icon: 800,
      temperature: {
        min: 12,
        max: 22,
      },
      wind: {
        speed: 23,
        direction: "s",
      },
      humidity: 30,
    },
  };

  it("renders correctly ", function () {
    const { asFragment } = render(
      <ForecastDetails forecast={validProps.forecast} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
