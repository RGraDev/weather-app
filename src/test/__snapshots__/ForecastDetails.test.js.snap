// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly  1`] = `
<DocumentFragment>
  <div
    class="forecast=details"
  >
    <div
      class="forecast-details__date"
    >
      Mon Apr 30 2018
    </div>
    <div
      class="forecast-details__description"
    >
      stub description
    </div>
    <div
      class="forecast-details__icon"
    >
      <img
        alt="weather icon"
        src="https://openweathermap.org/img/wn/04d@2x.png"
      />
    </div>
    <div
      class="forecast-details__min-temp"
    >
      12°C
    </div>
    <div
      class="forecast-details__max-temp"
    >
      22°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      30
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      23mph
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      S
    </div>
  </div>
</DocumentFragment>
`;
